Framework/Algorithms/test/PolarizationEfficiencyCorTest.h

```diff
@@ -486,10 +409,35 @@ class PolarizationEfficiencyCorTest : public CxxTest::TestSuite {
     return names;
   }
 
+  std::shared_ptr<PolarizationEfficiencyCor> createAlgorithm(const std::string &efficiencyMethod) {
+    auto alg = std::make_shared<PolarizationEfficiencyCor>();
+    alg->setRethrows(true);
+    alg->initialize();
+    alg->setProperty("OutputWorkspace", OUTPUT_GRP_NAME);
+    alg->setProperty("Efficiencies", createEfficiencies(efficiencyMethod));
+    return alg;
+  }
+
+  void setAlgorithmProperties(const std::shared_ptr<PolarizationEfficiencyCor> &alg, const std::string &method = "",
+                              const std::string &analysisMethod = "") {
+    alg->setProperty("OutputWorkspace", OUTPUT_GRP_NAME);
+    if (!method.empty()) {
```

Comment:
Could we move these two if statements into the `createAlgorithm` method (we'll need to add the `method` and `analysisMethod` parameters to `createAlgorithm` in order to to this)? Then we could remove this separate `setAlgorithmProperties` method, which should further simplify the tests as it seems to be needed in most of the tests.